src/steg86/binary.rs

```diff
@@ -647,6 +697,12 @@ impl Text {
             Err(anyhow!("couldn't find .text section"))
         }
     }
+
+    fn is_commutative(code: &Code) -> bool {
+        SEMANTIC_PAIRS
+            .iter()
+            .any(|&(c1, c2)| c1 == c2 && c1 == *code)
+    }
```

Comment:
Let's make this a `HashSet<Code>` inside a `lazy_static!` block, like `SUPPORTED_OPCODES`. Should be faster and won't require us duplicating each code inside `SEMANTIC_PAIRS` when they aren't really pairs.